hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CacheConfig.java

```diff
@@ -291,12 +291,36 @@ public void setCacheDataOnWrite(boolean cacheDataOnWrite) {
    * cacheIndexesOnWrite
    * cacheBloomsOnWrite
    */
-  public void enableCacheOnWrite() {
+  public void enableCacheOnWriteForCompactions() {
     this.cacheDataOnWrite = true;
     this.cacheIndexesOnWrite = true;
     this.cacheBloomsOnWrite = true;
   }
 
+  /**
+   * If hbase.rs.cachecompactedblocksonwrite configuration is set to true and
+   * 'totalCompactedFilesSize' is lower than 'cacheCompactedDataOnWriteThreshold',
+   * enables cache on write for below properties:
+   * - cacheDataOnWrite
+   * - cacheIndexesOnWrite
+   * - cacheBloomsOnWrite
+   *
+   * Otherwise, sets 'cacheDataOnWrite' only to false.
+   *
+   * @param totalCompactedFilesSize the total size of compacted files.
+   * @return true if the checks mentioned above pass and the cache is enabled, false otherwise.
+   */
+  public boolean enableCacheOnWriteForCompactions(long totalCompactedFilesSize) {
```

Comment:
Yeah, this logic was moved here from `HStore.createWriterInTmp` (as you noticed later on your following comment). Because now we override the writer creation in DirectStoreCompactor to not use this `HStore.createWriterInTmp`, we would need to duplicate this logic there, but since `CacheConfig` already has all the info required to decide on this, thought about moving it to `CacheConfig`. 

I reckon naming and cohesion is not been great, but I was just trying to concentrate on the original problem here of allowing compacting files getting created in the store directory, without the need for tmp dirs and renames.

Reply:
BTW, let me fix the javadoc description.

Reply:
The javadoc had already been updated to mention the cachecompactedblocksonwrite config property.

>Can always leave this old name as "deprecated" which calls a better-named version, if you think it would help  alleviate confusion.

This was not a method move/rename, just a logic that was previously embedded in `HStore.createWriterInTmp` that I decided to move here in its own method for reuse. The naming seems right for me, as it should only be called for compactions, not for other types of writes. I add this to javadoc.

Reply:
Cool, can resolve this then? (unless Stack has more to add)